### PR TITLE
Add coal mining

### PR DESCRIPTION
**Pull request by Rail-Runner**
Add coal as a possible drop when breaking solid rock.

If you can find some coal in a mine, you could probably mine your own too.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Should rock coal and charcoal even be the same thing?
If so. the current charcoal should be renamed to just "coal".

**Comment by Mecares** [comment]
As a geologist: Lignite, Stone coal and char coal are in my humble opinion similar enough to be treated all as coal for the purpose of a game like Cataclysm, despite being very different in reality.

**Comment by Rivet-the-Zombie** [comment]
Why not use coal instead? We've already got the stuff - it uses item id:`coal`.

**Comment by Mecares** [comment]
Can not find it in the item browser, is it used for anything?

**Comment by Coolthulhu** [comment]
Our "coal" is a large (72 volume) pallet of coal that requires a pickaxe and 8 fabrication to turn into charcoal.

**Comment by Rivet-the-Zombie** [comment]
I see no problem with folks getting the occasional coal-boulder while mining.

**Comment by Rivet-the-Zombie** [comment]
>Rename charcoal to coal, 

Why do this? It was fine already.

>and make solid rock drop coal boulder instead of coal.

This is good.

**Comment by Darkyhard** [comment]
> the current charcoal should be renamed to just "coal".
> don't do this; it doesn't need to be changed.

Two different people tell me to do mutually exclusive things. That's just odd.

My opinion is that "coal" may mean charcoal or stone coal, while "charcoal" may only mean charcoal, so it probably needs to be changed to be a correct name for both stone coal and charcoal.

**Comment by Rivet-the-Zombie** [comment]
If we're renaming 'charcoal' to 'coal' (though I've never known anyone to use the word coal to refer to the stuff made from wood) then the description needs to be altered as well. It currently describes the stuff as being made from wood very specifically.

**Comment by Darkyhard** [comment]
I can't think of a good description which doesn't mention if this batch of coal is stone coal or charcoal.

**Comment by Rivet-the-Zombie** [comment]
`Flammable black chunks of carbon-based material commonly used for cooking and heating.`

**Comment by kevingranade** [comment]
That's because it's intrinsic to what it is, please keep them seperate.
Feel free to clone charcoal and make it a 'lump of coal' item with similar
properties, but I don't want the black Rock you mine and the burnt stuff
you make out of wood to literally be the same item in the game.  Plus I'm
sure there are differences chemically.

**Comment by Darkyhard** [comment]
> I don't want the black Rock you mine and the burnt stuff you make out of wood to literally be the same item in the game

If we will ever have some uses for it that are exclusive to stone coal or charcoal, we may separate them at any moment, it would be quick. Such uses don't seem to exist yet, so I can't see the reasoning behind having two identical items with different names.

**Comment by Rivet-the-Zombie** [comment]
>If we will ever have some uses for it that are exclusive to stone coal or charcoal, we may separate them

I can actually think of two that we already have: charcoal water filters and lye. You *have to* use charcoal for those, as coal simply will not work for it.

Just make coal a subtype of charcoal so that it can be loaded into forges as such, and we can add it to recipes where it would be appropriate.

**Comment by Mecares** [comment]
Just call it Lignite or brown coal which is more similar to char coal and can be actually found in New England in low depths contrary to the average stone coal.

**Comment by Darkyhard** [comment]
Done. Charcoal and stone coal are now separated, and stone coal can be used in every recipe that uses charcoal, except lye powder, water filter and black gunpowder (because my other PR changes it too, so they would conflict if I change this recipe too).

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -2485,6 +2485,23 @@
...
+        "price" : 1000,
+        "name" : "coal",
+        "name_plural" : "coal",
+        "symbol" : "=",
+        "color" : "dark_gray",
+        "description" : "Flammable black chunks of carbon-based material commonly used for cooking and heating.",
+        "material" : "wood",
+        "volume" : 5,
+        "weight" : 220,
+        "phase" : "solid",
```
Comment (illi-kun) [inline data/json/items/ammo.json]: please remove ```"phase" : "solid",``` since it's default value anyway.

File: data/json/items/ammo.json [hunk 2]
```diff
@@ -2501,6 +2518,24 @@
...
+        "price" : 1000,
+        "name" : "coal pieces",
+        "name_plural" : "coal pieces",
+        "symbol" : "=",
+        "color" : "dark_gray",
+        "description" : "These are smaller pieces of coal used in smaller heating devices.",
+        "material" : "wood",
+        "volume" : 1,
+        "weight" : 22,
+        "phase" : "solid",
```
Comment (illi-kun) [inline data/json/items/ammo.json]: please remove ```"phase" : "solid",``` since it's default value anyway.

**Comment by illi-kun** [comment]
I'm not sure that the best file for coal is ```ammo.json``` (but I see why you've put it in this file - it's due to its type).

**Comment by Core0verload** [comment]
Why separate charcoal and regular coal? I don't get it.

**Comment by illi-kun** [comment]
@Core0verload they were separated because charcoal (byproduct from burnt wood) and coal (from mines) are not the same.

**Comment by Core0verload** [comment]
Does it justify adding another item that had to be placed alongside charcoal in every single crafting recipe because for the purposes of the game they _are _the same

**Comment by illi-kun** [comment]
> to be placed alongside charcoal in every single crafting recipe
That's not true, see previous discussion and code. Also, I see nothing wrong with such overlapping, more options for craft is generally good.

**Comment by Mecares** [comment]
Well the statement about charcoal being the only one usable for filters is wrong, activated carbon filters can be made from any coal, and lye is technically not made from char coal but from ashes remaining after burning wood and since brown coal is not much more than very old and compressed wood i do not see any problems using it for lye crafting.

**Comment by Core0verload** [comment]
Unless "more options" are just the same items under different names.

**Comment by Rivet-the-Zombie** [comment]
>activated carbon filters can be made from any coal

Which would require a separate activation step, and would leave us with something very similar to charcoal.

>lye is technically not made from char coal but from ashes remaining after burning wood

It's a simplification until we have ashes as an actual item you can hold in your hand.

**Comment by Mecares** [comment]
> Which would require a separate activation step, and would leave us with something very similar to charcoal.

The same methods have to be used to transform regular char coal into activated coal it is not like homemade char coal is more usable as activated coal than for example brown coal. I do not see how char coal qualifies as filter material and the other coals not.

> It's a simplification until we have ashes as an actual item you can hold in your hand.

The same simplification could be applied to brown coal.

**Comment by Cat09** [comment]
part of the ability of charcoal in a filter context is a matter of density, whereas brown coal is 

> very old and compressed wood 

anything that is mined is very likely to be rather compressed and dense, prohibiting water flow through it.

**Pull request merged**